app/Http/Controllers/Api/UsersController.php

```diff
@@ -0,0 +1,20 @@
+<?php namespace Symposium\Http\Controllers\Api;
+
+use Auth;
+use Talk;
+
+class UsersController extends BaseController
+{
+    /**
+     * Display all of the current user's talks
+     *
+     * @return Response
+     */
+    public function talks()
+    {
+        $talks = Talk::where('author_id', Auth::user()->id)
+            ->get();
+
+        return $this->quickJsonApiReturn($talks, 'talks');
+    }
+}
```

Comment:
Sticking methods outside of the standard REST stuff for the `users` resource in the `UsersController` gets messy fast. Creating a separate controller for nested resources lets you stick with the standard action names.

So you'd end up with 3 controllers in this case:

`UsersController` for regular REST stuff on `users`.
`TalksController` for regular REST stuff on `talks`.
`UserTalksController` for regular REST stuff on `talks` scoped to a `user`.

What you have here would end up being basically this:

`GET /users/{id}/talks`

```php
class UserTalksController
{
    public function index($userId)
    {
        return User::find($userId)->talks;
    }
}
```

...with some ACL elsewhere to verify the currently logged in user could access this person's talks. Notice how you get to just make it an `index` action instead of inventing a new one. Sticking to that pattern and convention helps lead to a nice and consistent code base :+1: